### PR TITLE
Use lazy_loader for pyspark

### PR DESCRIPTION
**Pull request by akhurana001**
# TL;DR
Use lazy_loaded for pyspark so that non-spark notebooks can run without pyspark installed.

Also added a dependency on `ipykernel` . Executions in my local venv were failing with a `No module named ipykernel_launcher` . 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Use lazy_loader for pyspark

## Tracking Issue
https://github.com/lyft/flyte/issues/488

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by kumare3** [comment]
Please update the description completely

File: setup.py [hunk 1]
```diff
@@ -50,6 +50,7 @@
         "urllib3>=1.22,<2.0.0",
         "wrapt>=1.0.0,<2.0.0",
         "papermill>=1.2.0",
+        "ipykernel>=5.0.0",
```
Comment (kumare3) [inline setup.py:53]: ?

File: flytekit/contrib/notebook/helper.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import os as _os
 
 import six as _six
-from pyspark import SparkConf, SparkContext
 
 from flytekit.common.types.helpers import pack_python_std_map_to_literal_map as _packer
 from flytekit.contrib.notebook.supported_types import notebook_types_map as _notebook_types_map
+from flytekit.plugins import pyspark as _pyspark
```
Comment (kumare3) [inline flytekit/contrib/notebook/helper.py:7]: this will work because only when we access pyspark.`x` will it load the real module?

File: setup.py (hunk 1 above)
Comment (akhurana001) [reply setup.py:53]: See the description.

File: flytekit/contrib/notebook/helper.py (hunk 2 above)
Comment (akhurana001) [reply flytekit/contrib/notebook/helper.py:7]: Yeah, that's my understanding of lazy_loading here: https://github.com/lyft/flytekit/blob/master/flytekit/plugins/__init__.py#L5
